### PR TITLE
Error page upgrades

### LINKED ISSUE #3200: Make Error Page Reactive to Online Status
### Observed behavior
Following various bugs or unknown errors, there is an appearance of a red box with json error. This format of displaying the error isn't particularly useful or informative to the user, and offers no information, it just breaks the current page the user is on. 

### Expected behavior
Some kind of friendly and informative way of conveying that an unexplainable error has occurred in Kolibri. 

### User-facing consequences
More informative and actionable error handling for unknown errors occurring in Kolibri. The more clarity we have in the design, the less put off users will be. We will also provide users with various options to report their error. 

**Design resources:**
[Spec and flow sheet](https://drive.google.com/open?id=1Dqx95ScIuVbcHWkVBzpr-jY8_bEs3UsTCDqZ3DiF3E4)
[Design of error display](https://drive.google.com/open?id=196xyQWHzGC2TbFXlNurRGvaYd8mOYhhm)
[Design of error report modal](https://drive.google.com/open?id=17HimanhkRkiGiiusLY0l4iOEUbfeI5jc)

[drive folder containing all of the above](https://drive.google.com/open?id=1Nk5B6MAYDND6rRYgCwOuZue5RSgf-emw)

### PR DESCRIPTION
**Pull request by DXCanas**
### Summary

New error page with associated screen shots:

error page

report error modal

Switched to large due to https://github.com/learningequality/kolibri/issues/3200#issuecomment-408276495

Here's what it looks like medium:
error report screencap, medium

### Reviewer guidance

Tried inducing errors, but found that the `CORE_SET_ERROR` mutation couldn't be called? Anyone else seeing this? @jonboiser might it have something to do with vuex modules?

### References

resolves #3200 

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If there are any front-end changes, before/after screenshots are included
- [x] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by benjaoming** [comment]
This sounds nice!

Can you provide a screen shot?

Why is it targeted for the `error-upgrades` branch?

File: kolibri/core/assets/src/views/app-body.vue [hunk 1]
```diff
@@ -10,9 +10,17 @@
...
-      <slot></slot>
+
+    <div
+      v-else
+      :style="bodyPadding"
+      class="body-wrapper"
+    >
+      <app-error v-if="error" />
+
+      <slot v-else>
```
Comment (christianmemije) [inline kolibri/core/assets/src/views/app-body.vue]: Same line

**Comment by indirectlylit** [comment]
@benjaoming see:

* https://docs.google.com/document/d/11K2_SrKfHFSnapAMcUDSu0Si3pD758gTm56UPw2BMc8/edit#heading=h.a15vlk954e6v
* https://drive.google.com/drive/folders/1Nk5B6MAYDND6rRYgCwOuZue5RSgf-emw

File: kolibri/core/assets/src/views/app-body.vue (hunk 1 above)
Comment (DXCanas) [reply kolibri/core/assets/src/views/app-body.vue]: Can we wait till this PR goes into develop to get this granular?

**Comment by DXCanas** [comment]
@benjaoming This is WIP work for a feature branch that @ralphiee22 @christianmemije and I are working on :) Not ready for merge into develop. Beginning to think opening a PR may have been the wrong move.

**Comment by benjaoming** [comment]
> Beginning to think opening a PR may have been the wrong move.

Ah no that's always fine if you're looking for inputs :) I was just asking because the final milestone was unclear and could guide the input that you might want -- which is why I was asking for a screenshot. Would seem like a good way to gather feedback.

But yeah, if you think *Reviewer guidance* cannot be filled in, then okay a PR may not be the way to go :)

**Comment by DXCanas** [comment]
Flagging: In an effort to induce an error, I noticed that called to the `CORE_SET_ERROR` mutation aren't actually being committed?

**Comment by DXCanas** [comment]
@khangmach Mind taking a quick look at those screencaps?

**Review by khangmach — APPROVED** [review]
very nice!!

**Comment by benjaoming** [comment]
@DXCanas This stuff looks great and is a great way of mixing detailed error reports with something that doesn't make users think that everything is wrong and they should move ten steps away from the computer :)

I know that this is perhaps a bit late for input, but the Refresh button:

1) Does it work if POST data was submitted? Can it loose other types of instance data being stored?
2) Does refreshing risk to loose the original error data?

**Comment by DXCanas** [comment]
@benjaoming 

> Does it work if POST data was submitted? Can it loose other types of instance data being stored?

It's a cache-busting URL-based refresh. So it'll lose any data _not_ encoded in the URL that was made anytime before or after the URL is loaded on browser. 

I think this behavior is expected, though. There are instances of requests being constructed based off of information being cached in the frontend. If the cache is out of date, and a request is made based on that faulty information, it'll likely result in a server-side error. In this case, we would want to refresh the page, rebuilding the state and cache.

> Does refreshing risk to loose the original error data?

Yes, at least on the frontend. The data error data is held in the store in the frontend, so if we want a record of the error it would be a matter of logging the request/response in a log or something. 

I think it's also important to note that the changes made in the PR have no effect on the way the error is being stored - frontend or backend.

**Pull request merged**